services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/AbstractFileBasedRegistryConfigProperties.java

```diff
@@ -31,6 +31,7 @@
     private boolean modificationEnabled = true;
     private boolean startEmpty = false;
     private int cacheMaxAge = DEFAULT_MAX_AGE_SECONDS;
+    private String[] hashAlgorithmsWhitelist = {};
```

Comment:
I would use a `HashSet`, rather than an array.

Reply:
An array is populated without extra work by springboot from a YAML list, or with comma separated values in an environment variable :) 

That said, the setter method could convert the array to a HashSet.